src/swarm/neo/util/AcquiredResources.d

```diff
@@ -14,6 +14,113 @@ module swarm.neo.util.AcquiredResources;
 
 import ocean.transition;
 
+private struct WrappedArray ( T )
```

Comment:
Note: the name of the struct is chosen pretty much at random. We can probably call it something better.